src/measure_theory/measure/measure_space.lean

```diff
@@ -2601,6 +2601,117 @@ begin
   exact (measure_mono (inter_subset_right _ _)).trans_lt (measure_spanning_sets_lt_top _ _),
 end
 
+/-- A set in a σ-finite space has zero measure if and only if its intersection with
+all members of the countable family of finite measure spanning sets has zero measure. -/
+lemma forall_measure_inter_spanning_sets_eq_zero
+  [measurable_space α] {μ : measure α} [sigma_finite μ] (s : set α) :
+  (∀ n, μ (s ∩ (spanning_sets μ n)) = 0) ↔ μ s = 0 :=
+begin
+  split,
+  { intros h,
+    rw (show s = ⋃ n, (s ∩ (spanning_sets μ n)),
+        by rw [← inter_Union, Union_spanning_sets, inter_univ]),
+    exact measure_Union_null_iff.mpr h, },
+  { intros s_null n,
+    exact le_antisymm (le_trans (measure_mono (inter_subset_left _ _)) s_null.le) (zero_le _), },
+end
+
+/-- A set in a σ-finite space has positive measure if and only if its intersection with
+some member of the countable family of finite measure spanning sets has positive measure. -/
+lemma exists_measure_inter_spanning_sets_pos
+  [measurable_space α] {μ : measure α} [sigma_finite μ] (s : set α) :
+  (∃ n, 0 < μ (s ∩ (spanning_sets μ n))) ↔ 0 < μ s :=
+begin
+  rw ← not_iff_not,
+  simp only [not_exists, not_lt, nonpos_iff_eq_zero],
+  exact forall_measure_inter_spanning_sets_eq_zero s,
+end
+
+/-- If the union of disjoint measurable sets has finite measure, then there are only
+finitely many members of the union whose measure exceeds any given positive number. -/
+lemma finite_const_le_meas_of_disjoint_Union {ι : Type*} [measurable_space α] {μ : measure α}
```

Comment:
The main part of this proof should be a result about `tsum` on `ennreal`: if you have a tsum on ennreal that is finite, then there are finitely many terms above `ε`. I feel like we should have that already somewhere.

Reply:
I found [`summable.tendsto_cofinite_zero`](https://leanprover-community.github.io/mathlib_docs/topology/algebra/infinite_sum.html#summable.tendsto_cofinite_zero). To me it would look useful to have also the more explicit statement about the finiteness of the set of indices for which the summands exceed a given positive level.

I will try to rewrite using this.

Reply:
In fact, the summands here are `ennreal`-valued (measures), and so `summable` is automatic by [`ennreal.summable`](https://leanprover-community.github.io/mathlib_docs/topology/instances/ennreal.html#ennreal.summable). Therefore [`summable.tendsto_cofinite_zero`](https://leanprover-community.github.io/mathlib_docs/topology/algebra/infinite_sum.html#summable.tendsto_cofinite_zero) does not apply here. Maybe one needs `ennreal`-specific versions, with `summable` hypothesis replaced by `∑' i, a i < ∞`.

Reply:
> The main part of this proof should be a result about `tsum` on `ennreal`: if you have a tsum on ennreal that is finite, then there are finitely many terms above `ε`. I feel like we should have that already somewhere.

As discussed, I don't believe we had this as such. I made #17588 which does that. The approach there is to translate measure theory results from counting measure to `tsum`, which may be a bit import-heavy, but I think it is way more convenient (I expect especially in the long run for more substantial manipulation of `tsum`s) than to use cardinalities API directly.

I'd be grateful if you can review that PR!